### PR TITLE
Refactored mock server service code

### PR DESCRIPTION
**Pull request by vipin230**
### :nut_and_bolt: Description: What code changed, and why?
Refactored Start and Stop code for mock server
Added  start and stop HTTP server

### :chains: Related Resources
https://chefio.atlassian.net/browse/CHEF-3442

### :athletic_shoe: How to Build and Test the Change

- Take the pull of this PR
- rebuild components/automate-cli/
- Install that cli build on the ubuntu or any linux supported by Automate
- Run chef-automate verify serve
- Curl request for start :- `curl --location 'http://<machine-ip>:7799/api/v1/start/mock-server' \
--header 'Content-Type: application/json' \
--data '{
  "port": <port>,
  "protocol": <protocol>,
  "cert":<cert-for-https>,
  "key":<key-for-https>
}'`
-Curl request for stop :- `curl --location 'http://<machine-ip>:7799/api/v1/stop/mock-server' \
--header 'Content-Type: application/json' \
--data '{
  "port": <port>,
  "protocol": <protocol>,
}'`

### :white_check_mark: Checklist

**All PRs** must tick these:

- [ ] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [ ] All commits signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

With occasional exceptions, all PRs **from Progress employees** must tick these:

- [ ] Is the code clear? *(complicated code or lots of comments--subdivide and use well-named methods, meaningful variable names, etc.)*
- [ ] Consistency checked? *(user notifications, user prompts, visual patterns, code patterns, variable names)*
- [ ] Repeated code blocks eliminated? *(adapt and reuse existing components, blocks, functions, etc.)*
- [ ] Spelling, grammar, typos checked? *(at a minimum use `make spell` in any component directory)*
- [ ] Code well-formatted? *(indents, line breaks, etc. improve rather than hinder readability)*

All PRs **from Progress employees** should tick these if appropriate:

- [ ] Tests added/updated? (all new code needs new tests)
- [ ] Docs added/updated? (all customer-facing changes)

*Please add a note next to any checkbox above if you are NOT ticking it.*

### :camera: Screenshots, if applicable
[start-stop-mockserver-using-curl-concurrently](https://progresssoftware.sharepoint.com/:v:/s/ChefCoreC/Ebj1Y9jJ8GVOsDHjez2fAlABJWdSPxkSClsumDt7LyJQqg?e=9E6Xau)
[start-stop-mockserver-api](https://progresssoftware.sharepoint.com/:v:/s/ChefCoreC/EU3xr1iptsdNghnab_u8cBEBGk4NDOIj6v9x5ZUUH1K-CQ?e=vbnTY1)

Screenshot 2023-06-09 at 4 17 30 PM
Screenshot 2023-06-09 at 4 18 25 PM

### CODE REVIEW

File: components/automate-cli/pkg/verifyserver/server/api/v1/startMockServer.go [hunk 1]
```diff
@@ -16,51 +16,34 @@ func (h *Handler) StartMockServer(c *fiber.Ctx) error {
...
-		h.Logger.Error(err)
-		return fiber.NewError(http.StatusBadRequest, "Invalid port number")
-	}
-
-	// Server is already running in the port
-	if err = validatePortAlreadyInUse(reqBody.Port, h); err != nil {
-		h.Logger.Error(err)
-		return fiber.NewError(http.StatusConflict, fmt.Sprintf(`"%s" server is already running on port %d`, reqBody.Protocol, reqBody.Port))
+	if !isValidPort(reqBody.Port) {
+		h.Logger.Error("Start mock-server request body contains invalid port number")
```
Comment (rishabhjhs) [inline components/automate-cli/pkg/verifyserver/server/api/v1/startMockServer.go:25]: Can we simplify this to 

return requestBodyPort >= 0 && requestBodyPort <= 65535

Comment (vipin230) [reply components/automate-cli/pkg/verifyserver/server/api/v1/startMockServer.go:25]: Done

**Review by rishabhjhs — APPROVED** [review]

**Review by ArvinthC3000 — APPROVED** [review]

**Review by shaik80 — APPROVED** [review]

**Review by bvtejaswi — APPROVED** [review]

**Review by Sunanda-Boorla — APPROVED** [review]

File: components/automate-cli/pkg/verifyserver/server/api/v1/startMockServer.go [hunk 2]
```diff
@@ -16,51 +16,31 @@ func (h *Handler) StartMockServer(c *fiber.Ctx) error {
...
+		if strings.Contains(err.Error(), "port unavailable") {
+			h.Logger.Error("Start mock-server request body contains unavailable port: ", reqBody.Port)
+			return fiber.NewError(fiber.StatusConflict, fmt.Sprintf(`Server is already running on port: %d`, reqBody.Port))
+		}
+		h.Logger.Error("Start mock-server request failed. Error: ", err.Error())
+		return fiber.NewError(fiber.StatusInternalServerError, err.Error())
 	}
 
-	return nil
+	return c.Status(fiber.StatusOK).JSON(response.BuildSuccessResponse("Server started successfully"))
```
[the remark this answers is not in the document]
Comment (vipin230) [reply components/automate-cli/pkg/verifyserver/server/api/v1/startMockServer.go]: Done

File: components/automate-cli/pkg/verifyserver/server/api/v1/stopmockserver.go [hunk 3]
```diff
@@ -17,55 +17,27 @@ func (h *Handler) StopMockServer(c *fiber.Ctx) error {
 	// If request body is invalid
 	if err != nil {
 		h.Logger.Error("Stop mock server request body parsing failed: ", err.Error())
-		return fiber.NewError(http.StatusBadRequest, "Invalid Body Request")
+		return fiber.NewError(fiber.StatusBadRequest, "Invalid Body Request")
```
[the remark this answers is not in the document]
Comment (vipin230) [reply components/automate-cli/pkg/verifyserver/server/api/v1/stopmockserver.go]: Done

**Review by vivek-yadav — APPROVED** [review]

**Review by atultherajput — APPROVED** [review]

**Review by vviveksharma — APPROVED** [review]

**Pull request merged**